modules/ding_provider/ding_provider.module

```diff
@@ -587,11 +590,16 @@ function ding_provider_get_pseudo_entity($ding_entity_id) {
   $ting_entity->ding_entity_id = $ding_entity_id;
   $ting_entity->provider_id = $local_id;
   $ting_entity->ding_entity_type = 'ding_entity';
-  $ting_entity->reply = new TingClientObject();
+  // TODO BBS-SAL: Clean out the direct property sets when done.
+  $ting_entity->reply = new TingPseudoObject();
   $ting_entity->reply->id = $ding_entity_id;
+  $ting_entity->reply->setDingId($ding_entity_id);
   $ting_entity->reply->localId = $local_id;
+  $ting_entity->reply->setLocalId($local_id);
```

Comment:
Ok, how about I do the following
- rename getDingId to getSearchId or maybe just getId() and then clearify in the docblock that this ID is used by the search provider to uniquely identify the object
- rename getLocalId to getSourceId and clearify in the docblock that this id is what the indexed source system used to identify the material.

Besides that it looks like I will eventually have to implement some kind of mapper-function in the search provider that will attempt to map from a source id (faust) to a search id for situations where the library system provider needs to go in that direction